qiskit/extensions/standard/barrier.py

```diff
@@ -33,6 +32,10 @@ def inverse(self):
         """Special case. Return self."""
         return Barrier(self.num_qubits)
 
+    def argument_expansion(self, qargs, cargs):
+        for qarg in zip(*qargs):
+            yield list(qarg), []
```

Comment:
Will this give us a separate barrier for every qarg, or one barrier across all qargs?

Reply:
Can you add a comment here what on why we want to `zip(*qargs)`?

Reply:
> Will this give us a separate barrier for every qarg, or one barrier across all qargs?

It should do the second. right?

Reply:
I simplified the approach in 94dac673